x-pack/solutions/search/plugins/search_playground/public/components/saved_playground/invalid_state_modal.tsx

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import {
+  EuiButton,
+  EuiModal,
+  EuiModalBody,
+  EuiModalFooter,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiText,
+  useGeneratedHtmlId,
+} from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import { SavedPlaygroundLoadErrors } from '../../types';
+
+export interface SavedPlaygroundInvalidStateModalProps {
+  errors: SavedPlaygroundLoadErrors;
+  onClose: () => void;
+}
+
+enum InvalidPlaygroundType {
+  MissingIndices,
+  MissingModel,
+  MissingIndicesAndModel,
+}
+
+export const SavedPlaygroundInvalidStateModal = ({
+  errors,
+  onClose,
+}: SavedPlaygroundInvalidStateModalProps) => {
+  const modalTitleId = useGeneratedHtmlId();
+  const errorsType: InvalidPlaygroundType =
+    errors.missingIndices.length > 0 && errors.missingModel !== undefined
+      ? InvalidPlaygroundType.MissingIndicesAndModel
+      : errors.missingModel !== undefined
+      ? InvalidPlaygroundType.MissingModel
+      : InvalidPlaygroundType.MissingIndices;
+  return (
+    <EuiModal
+      aria-labelledby={modalTitleId}
+      onClose={onClose}
+      data-test-subj="edit-playground-name-modal"
+    >
+      <EuiModalHeader>
+        <EuiModalHeaderTitle id={modalTitleId}>
+          {errorsType === InvalidPlaygroundType.MissingIndicesAndModel ? (
+            <FormattedMessage
+              id="xpack.searchPlayground.savedPlayground.invalidPlayground.title"
+              defaultMessage="Invalid playground values"
+            />
+          ) : errorsType === InvalidPlaygroundType.MissingModel ? (
+            <FormattedMessage
+              id="xpack.searchPlayground.savedPlayground.invalidModelConnector.title"
+              defaultMessage="AI Connector not found"
+            />
+          ) : (
+            <FormattedMessage
+              id="xpack.searchPlayground.savedPlayground.invalidDataSources.title"
+              defaultMessage="Invalid data sources"
+            />
+          )}
+        </EuiModalHeaderTitle>
+      </EuiModalHeader>
+      <EuiModalBody>
+        <EuiText>
+          {(errorsType === InvalidPlaygroundType.MissingIndicesAndModel ||
+            errorsType === InvalidPlaygroundType.MissingIndices) && (
+            <>
+              <p>
```

Comment:
as an elastic newbie, what benefit does <p> offer here? Are there cases where we want to add raw html elements instead of find something in EUI that fits the use case?

Reply:
[EuiText](https://eui.elastic.co/docs/components/display/text/) is a wrapper for for applying the correct typography styling to nested html elements and it gives explicit examples of using `<p>` <h1>` etc.

Another example where EUI does this is with the [EuiTitle](https://eui.elastic.co/docs/components/display/title/) component where you still us `hX` elements as a descendant of `EuiTitle`. If I remember correctly this is better for a11y.